marlowe-dashboard-client/src/WalletData/View.purs

```diff
@@ -36,48 +37,85 @@ newWalletCard newWalletNicknameKey wallets =
   in
     div
       [ classNames [ "flex", "flex-col" ] ]
-      [ h2_
+      [ p
+          [ classNames [ "mb-1" ] ]
           [ text "Create new contact" ]
-      , input
-          [ type_ InputText
-          , classNames $ toggleWhen (mNicknameError == Nothing) "border-green" "border-red"
-          , placeholder "Nickname"
-          , value nickname
-          , onValueInput_ SetNewWalletNickname
+      , div
+          [ classNames $ [ "mb-1" ] <> (applyWhen (nickname /= "") Css.hasNestedLabel) ]
+          [ label
+              [ classNames $ Css.nestedLabel <> hideWhen (nickname == "") ]
```

Comment:
personally I like `null` in these 2 places
`hideWhen (null nickname)` and `applyWhen ((not null) nickname)`